app/helpers/url.py

```diff
@@ -13,5 +13,10 @@ def get_base_url():
         The base endpoint for this service, where it is currently being served (or accessed)
     """
     # we need to remove either the trailing slash of request.host_url
-    # or the prefix slash of ROUTE_PREFIX (otherwise there are two slashes in between)
-    return f"{request.host_url}{ROUTE_PREFIX[1:]}"
+    # or the prefix slash of SCRIPT_NAME (otherwise there are two slashes in between)
+    base_url = f"{request.host_url[:-1]}{SCRIPT_NAME}"
+    # if base_url ends with a /, this needs to be removed, as following parts of the route will
+    # also begin with a / when added. Otherwise there would be // in the route.
+    if base_url.endswith('/'):
+        base_url = base_url[:-1]
+    return base_url
```

Comment:
Don't use SCRIPT_NAME in code, that's an anti pattern.
Here there is some simplification through the code that needs to be done and instead of using this `get_base_url()` we should use `url_for()`. The only exception where `url_for` cannot be used is to generate the `template_url` fields. There we can either make use of `request.root_url` (see https://flask.palletsprojects.com/en/2.0.x/api/#flask.Request.root_url) or the `url_for()` passing dummies argument to template for and then use a string replace; e.g 
```python
url_for('colorized_icon', 
    icon_set_name="{icon_set_name}"
    icon_name="{icon_name}",
    scale="{scale}",
    red="{r}",
    green="{g}",
    blue="{b}"
)', _external=True)
```
As the `{}` are theoretically not allowed in URL, they might cause trouble with `url_for`, in that case simply use `~~` for `{` and `~` for `}` instead (e.g. `icon_set_name="~~icon_set_name~"`) and then do `.replace('~~', '{').replace('~', '}')`.
If url_for accept the `{}` character I think that the best solution, if not either using `request.root_url` or `url_for` with string replacement (doesn't matter which one, as both have cons)

Reply:
Good Point, now `request.roo_url` is used instead, to return the base_url.